go/enclave/enclave_test.go

```diff
@@ -0,0 +1,225 @@
+package enclave
+
+import (
+	"crypto/rand"
+	"encoding/json"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/crypto/ecies"
+	"github.com/obscuronet/go-obscuro/go/common"
+	"github.com/obscuronet/go-obscuro/go/config"
+	"github.com/obscuronet/go-obscuro/go/rpc"
+	"github.com/obscuronet/go-obscuro/go/wallet"
+	"github.com/obscuronet/go-obscuro/integration"
+	"github.com/obscuronet/go-obscuro/integration/datagenerator"
+
+	gethcommon "github.com/ethereum/go-ethereum/common"
+)
+
+const _testEnclavePublicKeyHex = "034d3b7e63a8bcd532ee3d1d6ecad9d67fca7821981a044551f0f0cbec74d0bc5e"
+
+var _enclavePubKey *ecies.PublicKey
+
+func init() { //nolint:gochecknoinits
```

Comment:
@otherview Is it more idiomatic to do this with a `TestMain`, like here? https://github.com/obscuronet/go-obscuro/blob/main/integration/walletextension/wallet_extension_test.go#L94

Reply:
I don't think TestMain would be required here as it's not trying to execute tests in the execution scope. The init is not running a test per se, just initializing a var.

Reply:
I prefer something like the following, but maybe you think that's worse?

```
var _enclavePubKey = loadPubKey()

func loadPubKey() *ecies.PublicKey {
	// fetch the usable enclave pub key
	enclPubECDSA, err := crypto.DecompressPubkey(gethcommon.Hex2Bytes(_testEnclavePublicKeyHex))
	if err != nil {
		panic(err)
	}

	return ecies.ImportECDSAPublic(enclPubECDSA)
}
```

Reply:
Discussed offline 👍